data/glsk/glsk-document-cse/src/test/java/com/farao_community/farao/data/glsk/cse/CseGlskDocumentImporterTest.java

```diff
@@ -0,0 +1,166 @@
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ */
+public class CseGlskDocumentImporterTest {
+    private static final double EPSILON = 1e-3;
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyImportManualGskBlocks() {
+        CseGlskDocument cseGlskDocument = CseGlskDocument.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        List<AbstractGlskPoint> list = cseGlskDocument.getGlskPoints("FR_MANUAL");
+        assertFalse(list.isEmpty());
+        assertEquals(1, list.size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().size());
+        assertEquals(2, list.get(0).getGlskShiftKeys().get(0).getRegisteredResourceArrayList().size());
+    }
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyConvertManualGskBlocks() {
+        Network network = Importers.loadNetwork("testCase.xiidm", getClass().getResourceAsStream("/testCase.xiidm"));
+        GlskDocument glskDocument = GlskDocumentImporters.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        Scalable manualScalable = glskDocument.getZonalScalable(network).getData("FR_MANUAL");
+
+        assertNotNull(manualScalable);
+        assertEquals(2000., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3000., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+
+        manualScalable.scale(network, 1000.);
+        assertEquals(2700., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3300., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+    }
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyImportReserveGskBlocks() {
+        CseGlskDocument cseGlskDocument = CseGlskDocument.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        List<AbstractGlskPoint> list = cseGlskDocument.getGlskPoints("FR_RESERVE");
+        assertFalse(list.isEmpty());
+        assertEquals(1, list.size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().size());
+        assertEquals(2, list.get(0).getGlskShiftKeys().get(0).getRegisteredResourceArrayList().size());
+    }
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyConvertReserveGskBlocks() {
+        Network network = Importers.loadNetwork("testCase.xiidm", getClass().getResourceAsStream("/testCase.xiidm"));
+        GlskDocument glskDocument = GlskDocumentImporters.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        Scalable reserveScalable = glskDocument.getZonalScalable(network).getData("FR_RESERVE");
+
+        assertNotNull(reserveScalable);
+        assertEquals(2000., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(2000., network.getGenerator("FFR2AA1 _generator").getTargetP(), EPSILON);
+
+        reserveScalable.scale(network, -900.);
```

Comment:
What if variation is positive here ? Is it sill working ?

Reply:
Yes, but the result would not be what is expected (it is normal, see the comment on UpDownScalable not yet available)